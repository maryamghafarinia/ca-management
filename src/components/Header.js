import PropTypes from 'prop-types'

const Header = ({ title }) => {

  return (
    <header className='header'>
      <h2 data-testid='h2HeaderTag'>{title}</h2>
      <img style={{ width: '100px', cursor: 'pointer' }} alt="Auto1 Logo" src="https://auto1-js-task-api--mufasa71.repl.co/images/logo.png"></img>
    </header>
  )
}

Header.defaultProps = {
  title: 'Car Management',
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
}

export default Header
