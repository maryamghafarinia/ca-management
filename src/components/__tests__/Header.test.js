import React from "react";
import { render, cleanup } from  '@testing-library/react';
import Header from "../Header";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

test('renders correctly', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

it("inserts text in h2", () => {
  const { getByTestId, getByText } = render(<Header title="title" />);

  expect(getByTestId("h2HeaderTag")).toHaveTextContent("title");
});