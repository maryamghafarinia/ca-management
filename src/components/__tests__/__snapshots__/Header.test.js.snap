// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h2
      data-testid="h2HeaderTag"
    >
      Car Management
    </h2>
    <img
      alt="Auto1 Logo"
      src="https://auto1-js-task-api--mufasa71.repl.co/images/logo.png"
      style="width: 100px; cursor: pointer;"
    />
  </header>
</DocumentFragment>
`;
